internal/object/object_binary.go

```diff
@@ -100,8 +100,8 @@ func (o *Object) String() string {
 }
 
 func (o *Object) IsSpatial() bool {
-	_, ok := o.geo().(geojson.Spatial)
-	return ok
+	_, ok := o.geo().(String)
```

Comment:
This change (and consequently, the move of this file due to `collection` package) was due to:
* Circle does not implement `geojson.Spatial` interface, yet we do want to store it in the geo tree
*  `geojson.Spatial` does not seem to be used as an interface to call methods in this project, only as a marker.

I went with explicitly marking that we are not interested in strings here.
Alternatively, could possibly implement `geojson.Spatial` in `Circle`, open to other ideas as welll...